### PR TITLE
cephfs-shell: changes to stderr and stdout messages

### PR DESCRIPTION
**Pull request by rishabh-d-dave**
* strerror in OSError in cephfs.pyx should also contain the actual error messages
* align perror signature with cmd2.Cmd.perror
* stop passing default values to perror args
* use perror within module instead of cmd2.Cmd.perror
* use poutput within module instead of cmd2.Cmd.poutput
* print exception's error messages instead of printing custom messages everywhere

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

**Comment by varshar16** [comment]
Please rebase. Otherwise the patch lgtm.

File: src/tools/cephfs/cephfs-shell [hunk 1]
```diff
@@ -58,8 +58,8 @@ def poutput(s, end='\n'):
     shell.poutput(s, end=end)
 
 
-def perror(msg):
-    shell.perror(msg, end='\n', apply_style=True)
+def perror(msg, end='\n', apply_style=True):
```
Comment (varshar16) [inline src/tools/cephfs/cephfs-shell]: It's better to remove ```end='\n', apply_style=True```. As it is not compatible with older cmd2 versions.

**Comment by rishabh-d-dave** [comment]
Tests locally and on teuthology ran fine - http://pulpito.front.sepia.ceph.com/rishabh-2019-09-16_08:54:03-fs-fs-pybind-errmsgs-distro-basic-smithi/

File: src/tools/cephfs/cephfs-shell (hunk 1 above)
Comment (rishabh-d-dave) [reply src/tools/cephfs/cephfs-shell]: But that would disable us from using these arguments in future. IMO, it's better to have them at the cost of not being compatible with older version.

OR, if there's a serious need for it, we can check for `cmd2.__version__` beforehand. But it is not necessary/vital to be compatible with older version, so I suppose we can just skip that.

File: src/tools/cephfs/cephfs-shell [hunk 2]
```diff
@@ -535,9 +533,8 @@ exists.')
                 try:
                     cephfs.mkdir(path, permission)
                 except libcephfs.Error:
-                    self.perror('directory missing in the path; '
```
Comment (varshar16) [inline src/tools/cephfs/cephfs-shell]: Before libcephfs.Error, add exception to catch `ObjectNotFound` error. 
For this exception let the error message be `directory missing in the path; you may want to pass the -p argument`.

**Comment by varshar16** [comment]
For errors outside the defined classes in python bindings. If we use `Error` class, it will give `AttributeError: 'Error' object has no attribute 'strerror'`. So wherever `strerror` is used call `OSError` instead of `Error` class.

File: src/tools/cephfs/cephfs-shell (hunk 2 above)
Comment (rishabh-d-dave) [reply src/tools/cephfs/cephfs-shell]: Right now it prints `error in mkdir y/z: No such file or directory` which IMO is good enough. coreutils `mkdir` prints similar message -
```
$ mkdir y/z
mkdir: cannot create directory ‘y/z’: No such file or directory
```

**Comment by rishabh-d-dave** [comment]
> For errors outside the defined classes in python bindings. If we use `Error` class, it will give `AttributeError: 'Error' object has no attribute 'strerror'`. So wherever `strerror` is used call `OSError` instead of `Error` class.

Nice point but then we might miss exceptions that inherit directly from class `Error`. Therefore, I've added a method that checks type of exception and prints the error message accordingly. We'll use that method instead whenever we want to print error message already embedded in exception objects.

**Comment by varshar16** [comment]
I am getting following error:
```
CephFS:~/>>> ls test
'cephfs.LibCephFS' object has no attribute 'OSError'
Traceback (most recent call last):
  File "../src/tools/cephfs/cephfs-shell", line 131, in ls
    with cephfs.opendir(path) as d:
  File "cephfs.pyx", line 761, in cephfs.LibCephFS.opendir
cephfs.ObjectNotFound: opendir failed: No such file or directory [Errno 2]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "../src/tools/cephfs/cephfs-shell", line 407, in onecmd
    res = Cmd.onecmd(self, line, **kwargs)
  File "/home/varsha/ceph/build/venv/lib/python3.7/site-packages/cmd2/cmd2.py", line 2171, in onecmd
    stop = func(statement)
  File "/home/varsha/ceph/build/venv/lib/python3.7/site-packages/cmd2/cmd2.py", line 276, in cmd_wrapper
    return func(cmd2_app, args)
  File "../src/tools/cephfs/cephfs-shell", line 774, in do_ls
    key=lambda item: item.d_name)
  File "../src/tools/cephfs/cephfs-shell", line 140, in ls
    print_exception_msg(e)
  File "../src/tools/cephfs/cephfs-shell", line 63, in print_exception_msg
    if isinstance(e, cephfs.OSError):
AttributeError: 'cephfs.LibCephFS' object has no attribute 'OSError' 
```

**Comment by rishabh-d-dave** [comment]
> I am getting the following error:

Fixed. It prints now `opendir failed: No such file or directory` now.

File: src/tools/cephfs/cephfs-shell (hunk 2 above)
Comment (varshar16) [reply src/tools/cephfs/cephfs-shell]: Here we want the user to know  about the `-p` option. For other errors we can have standard messages.

**Comment by rishabh-d-dave** [comment]
Testing on teuthology was fine - http://pulpito.front.sepia.ceph.com/rishabh-2019-09-30_03:38:07-fs-wip-rishabh-fs-pybind-errmsgs-distro-basic-smithi/

**Comment by rishabh-d-dave** [comment]
Just a rebase.

**Comment by varshar16** [comment]
jenkins retest this please

**Comment by rishabh-d-dave** [comment]
The failure look unrelated - http://pulpito.front.sepia.ceph.com/rishabh-2019-10-20_14:14:04-fs-wip-rishabh-fs-pybind-errmsgs-distro-basic-smithi/

**Pull request merged**